README.md

```diff
@@ -297,13 +348,17 @@ spec:
       ratioMetric:
         counter: true
         good:
-          source: datadog
-          queryType: query
-          query: sum:requests.error{*}
+          metricSource:
+            name: datadog-datasource
+            type: Datadog
+            spec:
+              query: sum:trace.http.request.hits.by_http_status{http.status_code:200}.as_count()
```

Comment:
Is there a situation in which a datasource would not have a query defined?

I'm not sure I understand why `query` was moved down from the metric and into a `metricSource.spec` section. `query` is now an un-schema'd field that may or may not be there, or can be called something completely different.

Though some types of query would be specific to a datasource, there are also many that are common across datasources. From this perspective it feels odd to have the query as part of the `metricSource`

Reply:
I can imagine when you want to use @slok 's Sloth you might not define queries but use its plugin support

Reply:
That makes sense, but couldn't `query` be an optional at the `metric` level and not `metricSource` level?

Reply:
When thinking about this approach it felt more natural to put `spec` at the `metricSource` level because the user might provide some connection-specific details e.g. authentication for this particular metric source. 

Actually, `query` is part of `spec` and it might be replaced with other values. When connecting to  **Lighstep** we might pass something like this:
```
metricSource:
  metricSourceRef: lighstep-datasource
  spec:
    streamID: random
    typeOfData: latency
```

Depending on the data source, we might use different values to specify the metric we want to query. With the example above we see that the `query` field is not always necessary.

Reply:
Thanks for the example of when `query` wouldn't be present.

Reply:
I realize I'm a bit late to the party but I'm wondering if it's obvious to the users that they can reference a `DataSource` using `metricSourceRef` and at the same time add some values with the `spec` part as in @davex98 example. It wasn't obvious for me and I had to dig into this PR.
If the concern is not only mine, I'll open a separate PR :)